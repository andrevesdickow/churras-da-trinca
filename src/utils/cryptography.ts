import crypto from 'crypto';
import { isEmpty } from 'lodash';

function getBytes(str: string) {
  const bytes = [];
  for (let i = 0; i < str.length; ++i) {
    bytes.push(str.charCodeAt(i));
  }
  return bytes;
}

/**
 * Criptografa um texto
 * @param textToEncrypt texto
 * @param secretKey chave secreta
 * @returns texto criptografado
 */
export function encrypt(textToEncrypt: string, secretKey: string) {
  if (isEmpty(textToEncrypt) || isEmpty(secretKey)) {
    return '';
  }

  const key = Buffer.from(getBytes(secretKey));
  const iv = Buffer.from([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);

  const cipher = crypto.createCipheriv('aes-128-cbc', key, iv);
  let encrypted = cipher.update(textToEncrypt);
  encrypted = Buffer.concat([encrypted, cipher.final()]);
  return encrypted.toString('base64');
}
