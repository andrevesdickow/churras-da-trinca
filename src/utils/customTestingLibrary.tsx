import '../app/global.css';
import React from 'react';
import { ToastContainer } from '@/components/Toast';
import { NextThemeProvider } from '@/providers/NextThemeProvider';
import { Provider as TooltipProvider } from '@radix-ui/react-tooltip';
import { render } from '@testing-library/react';

const customRender = (ui: React.ReactElement, { ...renderOptions } = {}) => render(
  <NextThemeProvider>
    <TooltipProvider>
      {ui}
      <ToastContainer />
    </TooltipProvider>
  </NextThemeProvider>,
  renderOptions
);

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
