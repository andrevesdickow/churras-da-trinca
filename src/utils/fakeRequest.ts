export function fakeRequest<T = any>(delay: number, value: T) {
  return new Promise((resolve) => setTimeout(resolve, delay, value));
}
