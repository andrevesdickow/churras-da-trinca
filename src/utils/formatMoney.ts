/**
 * Formata valor monetÃ¡rio para real brasileiro
 * @param value valor
 * @returns Valor formatado -> R$ 1.000,00
 */
export function formatMoney(value: number) {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL'
  }).format(value);
}
