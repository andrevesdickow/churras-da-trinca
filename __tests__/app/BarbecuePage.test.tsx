import React from 'react';
import BarbecuePage from '@/app/churras/page';
import { render, screen } from '@/utils/customTestingLibrary';

jest.mock('@tanstack/react-query', () => ({
  ...jest.requireActual('@tanstack/react-query'),
  useQuery: () => ({
    data: [{
      id: '1',
      date: '2023-12-25',
      dateFormatted: '25/12',
      description: 'Festa de Natal',
      priceWithDrink: 100,
      priceWithoutDrink: 50
    }],
    isFetching: false
  })
}));

describe('<BarbecuePage />', () => {
  it('should render correctly Barbecue Page', async () => {
    const { container } = render(<BarbecuePage />);

    expect(screen.getByText(/Adicionar churras/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should click create barbecue link correctly in Barbecue Page', () => {
    render(<BarbecuePage />);

    expect(screen.getByRole('link', { name: 'Adicionar churras' })).toHaveAttribute('href', '/churras/criar');
  });
});
