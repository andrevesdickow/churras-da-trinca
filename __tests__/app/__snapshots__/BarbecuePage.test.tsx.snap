// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarbecuePage /> should render correctly Barbecue Page 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="w-full grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-3 place-content-center gap-4 flex-col"
  >
    <a
      aria-label="Adicionar churras"
      class="text-amber-400 hover:text-amber-500 transition-colors"
      href="/churras/criar"
    >
      <div
        class="w-full h-full min-h-[160px] dark:text-slate-50 flex flex-col gap-6 p-6 rounded-lg shadow-xl bg-slate-200 dark:bg-slate-900 text-slate-800 hover:bg-slate-300 hover:dark:bg-slate-700 transition-colors justify-center"
      >
        <div
          class="flex justify-between flex-col items-center gap-4"
        >
          <img
            alt="Churras"
            data-nimg="1"
            decoding="async"
            height="70"
            loading="lazy"
            src="/icon-churras.svg"
            style="color: transparent;"
            width="70"
          />
          <span
            class="font-bold"
          >
            Adicionar churras
          </span>
        </div>
      </div>
    </a>
  </div>
</div>
`;
